snakedeploy/collect_files.py

```diff
@@ -0,0 +1,59 @@
+import sys
+from glob import glob
+import re
+import logging
+from collections import namedtuple
+
+import pandas as pd
+
+from snakedeploy.exceptions import UserError
+
+
+def collect_files(config_sheet_path: str):
+    config_sheet = pd.read_csv(config_sheet_path, sep="\t")
+    config_sheet["input_re"] = config_sheet["input_pattern"].apply(re.compile)
+
+    for item in sys.stdin:
+        item = item[:-1]  # remove newline
+
+        matches = list(
+            filter(
+                lambda match: match.match is not None, get_matches(item, config_sheet)
+            )
+        )
+
+        if not matches:
+            raise UserError(f"No input pattern in config sheet matches {item}.")
+        elif len(matches) > 1:
+            raise UserError(f"Item {item} matches multiple input patterns.")
+        else:
```

Comment:
For this third else, we can probably remove it and move the rest of the function one indentation out. If either of the first two blocks are triggered, raising the error will stop execution (so I don't think we would need the else?) E.g., in this small example we never get to the print statement:

```python
from snakedeploy.exceptions import UserError

if 2==1:
   print("Case 1")

# This one will trigger
elif 1==1:
    raise UserError('blah')

# We never get here
print('NO')
```